lib/storage/search.go

```diff
@@ -296,6 +299,33 @@ func (sq *SearchQuery) GetTimeRange() TimeRange {
 	}
 }
 
+// Exec calls cb for with marshaled requestData for each tenant in sq.
+func (sq *SearchQuery) Exec(qt *querytracer.Tracer, cb func(qt *querytracer.Tracer, requestData []byte, t TenantToken) any) []any {
```

Comment:
It's better to move `Exec` function to the app/vmselect/netstorage package. 

It should accept search query as argument.